gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/sql/execute/SnappyUpdateDeleteResultSet.java

```diff
@@ -40,7 +40,7 @@ public int modifiedRowCount() {
       rowCount = 0;
       try {
         ExecRow row;
-        while ((row = getNextRow()) != null) {
+        while ((row = getNextRow()) != null && row.nColumns() > 0) {
```

Comment:
Currently insert-select return rows without any column. Once that is fixed we can make change you are suggesting. Will add a comment here.